### ISSUE TITLE
Debug page does not work when logged in to an IdP

### ISSUE THREAD

**Issue by phavekes** [OpenConext/OpenConext-engineblock#1648]
[This issue is imported from pivotal - Originaly created at Sep 2, 2020 by Thijs Kinkhorst](https://www.pivotaltracker.com/story/show/174627328)

Starting from EB 6.3.1.

The debug page  https://engine.test2.surfconext.nl/authentication/sp/debug (as linked from the engine homepage) works fine in a fresh browser/private browser.

When already logged into another SP, it fails with the "no_idps" error. So to reproduce:
1. Start browser session
2. Log in to e.g. Profile or Teams
3. Then go to engine homepage and click the debug link
4. Get presented with the "No IdPs" error.

Some logging follows.

```
Sep  2 17:17:36 t06.ams.surfconext.nl EBLOG[27970]: [2020-09-02 17:17:36] app.INFO: Handling incoming request: GET /authentication/sp/debug {"session_id":null,"request_id":"5f4fb7908a19a"} []
Sep  2 17:17:36 t06.ams.surfconext.nl EBLOG[27970]: [2020-09-02 17:17:36] request.INFO: Matched route "authentication_sp_debug". {"session_id":null,"request_id":"5f4fb7908a19a"} {"route":"authentication_sp_debug","route_parameters":{"_controller":"engineblock.controller.authentication.debug:debugSpConnectionAction","domain":"test2.surfconext.nl","_route":"authentication_sp_debug"},"request_uri":"https://engine.test2.surfconext.nl/authentication/sp/debug","method":"GET"}
Sep  2 17:17:36 t06.ams.surfconext.nl EBLOG[27970]: [2020-09-02 17:17:36] security.INFO: Populated the TokenStorage with an anonymous Token. {"session_id":null,"request_id":"5f4fb7908a19a"} []
Sep  2 17:17:36 t06.ams.surfconext.nl EBLOG[27970]: [2020-09-02 17:17:36] app.INFO: Calling service \'debugSingleSignOnService\' {"session_id":"47ck2a09frne1ojg5gnv4irjhl","request_id":"5f4fb7908a19a"} []
Sep  2 17:17:36 t06.ams.surfconext.nl EBLOG[27970]: [2020-09-02 17:17:36] app.INFO: Getting request... {"session_id":"47ck2a09frne1ojg5gnv4irjhl","request_id":"5f4fb7908a19a"} []
Sep  2 17:17:36 t06.ams.surfconext.nl EBLOG[27970]: [2020-09-02 17:17:36] app.INFO: Created debug SAML request {"session_id":"47ck2a09frne1ojg5gnv4irjhl","request_id":"5f4fb7908a19a"} {"saml_request":"<?xml version=\"1.0\"?>\n<samlp:AuthnRequest xmlns:samlp=\"urn:oasis:names:tc:SAML:2.0:protocol\" xmlns:saml=\"urn:oasis:names:tc:SAML:2.0:assertion\" ID=\"CORTOecb316cd941ab45b003553218bd6e5c88c73ea1c\" Version=\"2.0\" IssueInstant=\"2020-09-02T15:17:36Z\"><saml:Issuer>https://engine.test2.surfconext.nl/authentication/sp/metadata</saml:Issuer></samlp:AuthnRequest>\n"}
Sep  2 17:17:36 t06.ams.surfconext.nl EBLOG[27970]: [2020-09-02 17:17:36] app.INFO: Fetching service provider matching request issuer \'https://engine.test2.surfconext.nl/authentication/sp/metadata\' {"session_id":"47ck2a09frne1ojg5gnv4irjhl","request_id":"5f4fb7908a19a"} []
Sep  2 17:17:36 t06.ams.surfconext.nl EBLOG[27970]: [2020-09-02 17:17:36] app.INFO: Determining whether service provider in chain requires additional logging {"session_id":"47ck2a09frne1ojg5gnv4irjhl","request_id":"5f4fb7908a19a"} []
Sep  2 17:17:36 t06.ams.surfconext.nl EBLOG[27970]: [2020-09-02 17:17:36] app.INFO: No additional logging required {"session_id":"47ck2a09frne1ojg5gnv4irjhl","request_id":"5f4fb7908a19a"} []
Sep  2 17:17:36 t06.ams.surfconext.nl EBLOG[27970]: [2020-09-02 17:17:36] app.INFO: Request lists scoped IdPs {"session_id":"47ck2a09frne1ojg5gnv4irjhl","request_id":"5f4fb7908a19a"} {"request_idps":[]}
Sep  2 17:17:36 t06.ams.surfconext.nl EBLOG[27970]: [2020-09-02 17:17:36] app.INFO: Remembered last used IDP - limiting scope to last selection {"session_id":"47ck2a09frne1ojg5gnv4irjhl","request_id":"5f4fb7908a19a"} []
Sep  2 17:17:36 t06.ams.surfconext.nl EBLOG[27970]: [2020-09-02 17:17:36] app.INFO: 0 candidate IdPs after scoping {"session_id":"47ck2a09frne1ojg5gnv4irjhl","request_id":"5f4fb7908a19a"} {"idps":[]}
Sep  2 17:17:36 t06.ams.surfconext.nl EBLOG[27970]: [2020-09-02 17:17:36] app.NOTICE: No Identity Provider {"session_id":"47ck2a09frne1ojg5gnv4irjhl","request_id":"5f4fb7908a19a"} []
Sep  2 17:17:36 t06.ams.surfconext.nl EBLOG[27970]: [2020-09-02 17:17:36] engineblock.ERROR: No candidate IdPs found | -> Redirecting to feedback page {"session_id":"47ck2a09frne1ojg5gnv4irjhl","request_id":"5f4fb7908a19a"} {"exception":{"severity":"ERROR","location":"/opt/openconext/OpenConext-engineblock-6.3.1/library/EngineBlock/Corto/Module/Service/SingleSignOn.php:178","userId":null,"idp":null,"sp":null,"details":"EngineBlock_Corto_Module_Service_SingleSignOn_NoIdpsException: No candidate IdPs found\n#0 /opt/openconext/OpenConext-engineblock-6.3.1/library/EngineBlock/Corto/Module/Services.php(74): EngineBlock_Corto_Module_Service_SingleSignOn->serve(\'debugSingleSign...\', Object(Symfony\\Component\\HttpFoundation\\Request))\n#1 /opt/openconext/OpenConext-engineblock-6.3.1/library/EngineBlock/Corto/ProxyServer.php(373): EngineBlock_Corto_Module_Services->serve(\'debugSingleSign...\')\n#2 /opt/openconext/OpenConext-engineblock-6.3.1/library/EngineBlock/Corto/Adapter.php(329): EngineBlock_Corto_ProxyServer->serve(\'debugSingleSign...\', \'\')\n#3 /opt/openconext/OpenConext-engineblock-6.3.1/library/EngineBlock/Corto/Adapter.php(77): EngineBlock_Corto_Adapter->_callCortoServiceUri(\'debugSingleSign...\')\n#4 /opt/openconext/OpenConext-engineblock-6.3.1/src/OpenConext/EngineBlockBundle/Controller/DebugController.php(56): EngineBlock_Corto_Adapter->debugSingleSignOn()\n#5 /opt/openconext/OpenConext-engineblock-6.3.1/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php(151): OpenConext\\EngineBlockBundle\\Controller\\DebugController->debugSpConnectionAction()\n#6 /opt/openconext/OpenConext-engineblock-6.3.1/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php(68): Symfony\\Component\\HttpKernel\\HttpKernel->handleRaw(Object(Symfony\\Component\\HttpFoundation\\Request), 1)\n#7 /opt/openconext/OpenConext-engineblock-6.3.1/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php(200): Symfony\\Component\\HttpKernel\\HttpKernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)\n#8 /opt/openconext/OpenConext-engineblock-6.3.1/web/app.php(19): Symfony\\Component\\HttpKernel\\Kernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request))\n#9 {main}","message_prefix":null}}
Sep  2 17:17:36 t06.ams.surfconext.nl EBLOG[27970]: [2020-09-02 17:17:36] app.NOTICE: An error was caught {"session_id":"47ck2a09frne1ojg5gnv4irjhl","request_id":"5f4fb7908a19a"} []
Sep  2 17:17:36 t06.ams.surfconext.nl Apache-EB: 145.100.191.114 - - [02/Sep/2020:17:17:36 +0200] "GET /authentication/sp/debug HTTP/1.1" 302 372 "https://engine.test2.surfconext.nl/" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:80.0) Gecko/20100101 Firefox/80.0" "195.169.126.164"
Sep  2 17:17:36 t06.ams.surfconext.nl EBLOG[27974]: [2020-09-02 17:17:36] app.INFO: Handling incoming request: GET /authentication/feedback/no-idps {"session_id":null,"request_id":"5f4fb790aa5ea"} []
```

**Comment by phavekes** [comment]
I have confirmed that it does _not_ happen in 6.3.0. (Thijs Kinkhorst - Sep 2, 2020)

**Comment by phavekes** [comment]
I was able to reproduce this behaviour. And is caused by the SAML2 upgrade.

The CachedResponses are used to find the previously used IdP. This cache was previously filled with the entity Ids (string) of the SP/IdP pair. Now an Issuer value object was saved. Causing issues when using that value as a string.

Fixed in: https://github.com/OpenConext/OpenConext-engineblock/pull/889 (Michiel Kodde - Sep 3, 2020)
